pkg/api/v1/load_balancers.go

```diff
@@ -0,0 +1,281 @@
+		r.logger.Debugw("path param", "load_balancer_id", loadBalancerID)
+	}
+	// query params
+	queryParams := []string{"load_balancer_size", "load_balancer_type", "ip_addr", "location_id", "display_name"}
+
+	qpb := echo.QueryParamsBinder(c)
+
+	for _, qp := range queryParams {
+		mods = queryParamsToQueryMods(qpb, qp, mods)
+
+		if len(c.QueryParam(qp)) > 0 {
+			r.logger.Debugw("query param", "query_param", qp, "param_vale", c.QueryParam(qp))
+		}
+	}
+
+	err = qpb.BindError()
+	if err != nil {
+		return nil, err
+	}
+
+	return mods, nil
+}
+
+// loadBalancerGet returns a load balancer for a tenant by ID
+func (r *Router) loadBalancerGet(c echo.Context) error {
+	ctx, span := tracer.Start(c.Request().Context(), "loadBalancerGet")
+	defer span.End()
+
+	span.SetAttributes(attribute.String("route", "loadBalancerGet"))
+
+	mods, err := r.loadBalancerParamsBinding(c)
+	if err != nil {
+		r.logger.Errorw("bad request", "error", err)
+		return c.JSON(http.StatusBadRequest, v1BadRequestResponse(err))
+	}
+
+	lbs, err := models.LoadBalancers(mods...).All(ctx, r.db)
+	if err != nil {
+		return err
+	}
+
+	switch len(lbs) {
+	case 0:
+		return c.JSON(http.StatusNotFound, v1NotFoundResponse())
+	case 1:
+		return c.JSON(http.StatusOK, v1LoadBalancer(lbs[0]))
+	default:
+		return c.JSON(http.StatusOK, v1LoadBalancerSlice(lbs))
+	}
+}
+
+// loadBalancerGetByID returns a load balancer for a tenant by ID
+func (r *Router) loadBalancerGetByID(c echo.Context) error {
+	ctx, span := tracer.Start(c.Request().Context(), "loadBalancerGetByID")
+	defer span.End()
+
+	span.SetAttributes(attribute.String("route", "loadBalancerGetByID"))
+
+	// Look up the load balancer by ID from the path
+	lb, err := models.FindLoadBalancer(ctx, r.db, c.Param("load_balancer_id"))
+	if err != nil {
+		return err
+	}
+
+	tenantID, err := parseTenantID(c)
+	if err != nil {
+		r.logger.Errorw("bad request", "error", err)
+		return c.JSON(http.StatusBadRequest, v1BadRequestResponse(err))
+	}
+
+	// Ensure the tenant ID matches the tenant ID in the path
+	// If not, return a 404. This prevents a tenant from accessing another tenant's load balancer
+	if lb.TenantID != tenantID {
+		r.logger.Errorw("not found", "db_tenant_id", lb.TenantID, "path_tenant_id", tenantID)
+		return c.JSON(http.StatusNotFound, v1NotFoundResponse())
+	}
+
+	return c.JSON(http.StatusOK, v1LoadBalancer(lb))
+}
+
+// loadBalancerCreate creates a new load balancer for a tenant
+func (r *Router) loadBalancerCreate(c echo.Context) error {
+	ctx, span := tracer.Start(c.Request().Context(), "loadBalancerCreate")
+	defer span.End()
+
+	span.SetAttributes(attribute.String("route", "loadBalancerCreate"))
+
+	type input struct {
+		DisplayName      string `json:"display_name"`
+		LoadBalancerSize string `json:"load_balancer_size"`
+		LoadBalancerType string `json:"load_balancer_type"`
+		IPAddr           string `json:"ip_addr"`
+		LocationID       string `json:"location_id"`
+	}
```

Comment:
I would probably define this outside of the function and export it

Reply:
This was something discussed I discussed with @nicolerenee , since there isn't a client to `loadbalanacerapi` it isn't what we want to do. Everything will communicate through the API Gateway